pipeline/alert.go

```diff
@@ -560,6 +565,84 @@ type HipChatHandler struct {
 	Token string
 }
 
+// Send the alert to Alerta.
+//
+// Example:
+//    [alerta]
+//      enabled = true
+//      url = "https://alerta.yourdomain"
+//      token = "9hiWoDOZ9IbmHsOTeST123ABciWTIqXQVFDo63h9"
+//      environment = "Production"
+//      origin = "Kapacitor"
+//
+// In order to not post a message every alert interval
+// use AlertNode.StateChangesOnly so that only events
+// where the alert changed state are posted to the room.
+//
+// Example:
+//    stream...
+//         .alert()
+//             .alerta()
+//                 .resource('Hostname or service')
+//                 .event('Something went wrong')
+//
+// Send alerts to Alerta. Alerta requires a resource and event description.
+//
+// Example:
+//    stream...
+//         .alert()
+//             .alerta()
+//                 .resource('Hostname or service')
+//                 .event('Something went wrong')
+//                 .environment('Development')
+//                 .status('Open')
+//                 .group('Dev. Servers')
+//
+// Send alerts to Alerta. Alerta accepts detailed alert information.
+// tick:property
+func (a *AlertNode) Alerta() *AlertaHandler {
+	alerta := &AlertaHandler{
+		AlertNode: a,
+	}
+	a.AlertaHandlers = append(a.AlertaHandlers, alerta)
+	return alerta
+}
+
+// tick:embedded:AlertNode.Alerta
+type AlertaHandler struct {
+	*AlertNode
+
+	// Alerta authentication token.
+	// If empty uses the token from the configuration.
+        Token string
+
+	// Alerta resource.
+        // This is a required field.
+        Resource string
+
+	// Alerta event.
+	// This is a required field.
+        Event string
+
+	// Alerta environment.
+	// If empty uses the environment from the configuration.
+        Environment string
+
+	// Alerta status.
+        Status string
```

Comment:
What status are you using internally?

Kapacitor is about generating alerts and doesn't (at least not yet) want to deal with managing alert workflows. This is why Kapacitor wants to integrate with many third party tools like Alerta that do that for you. So to me it seems logical to have the alert status be in the 'open' state when Kapacitor creates it and then in the 'closed' state once it resolves. The other states are about interacting with the alert which you cannot do with Kapacitor.